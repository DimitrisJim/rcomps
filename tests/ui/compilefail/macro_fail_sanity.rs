// Check cases that compilefail.
//
// See https://rustc-dev-guide.rust-lang.org/tests/adding.html#error-annotations
// for annotating the snippets with the expected error.
// ignore-windows
extern crate rcomps;
use rcomps::comp;

/// Small sanity checks.
fn test_sanity(){
    comp!([20]);                       //~ ERROR Unable to parse expression.
    comp!([for n in 1..2 30]);         //~ ERROR Unable to parse expression.
    // todo: look into this.
    comp!([for n in n => n]);          //~ ERROR cannot find value `n` in this scope [E0425]

    comp!([for n in n]);               //~ ERROR Unable to parse expression.
    comp!([for n in 1..2 => 30; if]);  //~ ERROR Unable to parse expression.

    // Error for solo trailing greek question mark:
    // comp!([for i in 1..2 => 0;]);
    // comp!({for u in 1..2 => 0;});
    // comp!({for u in 1..2 => 0, 0;});
}

fn main() {}

