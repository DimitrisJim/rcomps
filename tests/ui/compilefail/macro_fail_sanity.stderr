error: no rules expected the token `[`
  --> $DIR/macro_fail_sanity.rs:11:5
   |
11 |     comp!([20]);                       //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^ no rules expected this token in macro call
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `]`
  --> $DIR/macro_fail_sanity.rs:12:5
   |
12 |     comp!([for n in 1..2 30]);         //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `]`
  --> $DIR/macro_fail_sanity.rs:16:5
   |
16 |     comp!([for n in n]);               //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `]`
  --> $DIR/macro_fail_sanity.rs:17:5
   |
17 |     comp!([for n in 1..2 => 30; if]);  //~ ERROR Unable to parse expression.
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find value `n` in this scope
  --> $DIR/macro_fail_sanity.rs:14:21
   |
14 |     comp!([for n in n => n]);          //~ ERROR cannot find value `n` in this scope [E0425]
   |                     ^ not found in this scope

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0425`.
